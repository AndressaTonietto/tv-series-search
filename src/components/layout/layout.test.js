import "@testing-library/jest-dom";

import * as router from "react-router";

import { MemoryRouter, useNavigate } from "react-router-dom";
import { fireEvent, render, screen } from "@testing-library/react";

import Layout from "./layout";
import renderer from "react-test-renderer";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: jest.fn(),
}));

describe("Layout", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Layout />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("navigates to home page when button is clicked", () => {
    const mockNavigate = jest.fn();
    useNavigate.mockReturnValue(mockNavigate);

    render(
      <MemoryRouter>
        <Layout />
      </MemoryRouter>
    );

    fireEvent.click(screen.getByText("TV Series Search"));
    expect(mockNavigate).toHaveBeenCalledWith("/");
  });
});
