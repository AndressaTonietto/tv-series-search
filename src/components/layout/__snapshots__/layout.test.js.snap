// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div
  className="min-h-screen"
>
  <div
    className="navbar bg-base-100"
  >
    <button
      className="btn btn-ghost text-2xl"
      onClick={[Function]}
    >
      TV Series Search
    </button>
  </div>
  <div
    className="p-2"
  />
</div>
`;
